### PR TITLE
seek.rb add search on classification

### LINKED ISSUE #2145: `seek.rb` add search on classification
Screen Shot 2023-03-31 at 11 44 22 pm

### PR DESCRIPTION
**Pull request by Dxuian**
resolves #2145 
added cateogry feature to the existing interface via web scraping 
image

### CODE REVIEW

**Comment by Dxuian** [comment]
(i  may have forgot to make a new branch 🙏🙏 please review and merge my [other ](https://github.com/slurpcode/slurp/pull/2785) pr before this one)

**Comment by BaseMax** [comment]
A review message from John:

> https://github.com/slurpcode/slurp/pull/2788/files#diff-910be1d5d0467f95668277ac980e47425d4571eeb06f0012038dd84133d80bb2  hey see the JSON data file added in this PR ?  I think it is a good start but I believe the second part of this task for another PR is to create a cron job the runs once per week that runs on GitHub Actions and it would be another mining script that hits seek dot com and scrapes just the "categories" and updates the JSON file with a bot PR ?  Otherwise how to keep these categories up to date ?

**Comment by jbampton** [comment]
Hey @Dxuian can you please fix the conflicts ?

**Comment by Dxuian** [comment]
> Hey @Dxuian can you please fix the conflicts ?

yes in my next commit i will first be fixing the confilcts 
ive also managed to make the script that web scrapes for the json file

**Comment by Dxuian** [comment]
@jbampton  ive managed to resolve the old conflict and am done with the web scraping part of the json maker

**Comment by jbampton** [comment]
Hey @Dxuian can you please fix these 2 issues:

https://app.deepsource.com/gh/slurpcode/slurp/run/71869dcc-e856-455a-86ed-1112d4f02f33/ruby/RB-LI1073?listindex=0

**Comment by BaseMax** [comment]
This branch has conflicts that must be resolved
Use the [web editor](https://github.com/slurpcode/slurp/pull/2788/conflicts) or the  to resolve conflicts.
Conflicting files
seek/seek.rb

**Comment by Dxuian** [comment]
> Hey @Dxuian can you please fix these 2 issues:
> 
> https://app.deepsource.com/gh/slurpcode/slurp/run/71869dcc-e856-455a-86ed-1112d4f02f33/ruby/RB-LI1073?listindex=0

aplogies some test code wiggled thorugh to the pr ill fix it in new pr  just now 
(ignore the large number of commits im about to make ive never used github actions so im learning as i go)

**Comment by jbampton** [comment]
I can always squash merge all your commits into one commit on merge

File: .github/workflows/weekly_run.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
 
       - name: Run t.rb
         run: |
-          ruby t.rb
+          ruby ./e/OS/slurp/t.rb
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml]: cd seek; ruby t.rb 

??

File: .github/workflows/weekly_run.yml [hunk 2]
```diff
@@ -17,7 +17,11 @@ jobs:
       - name: Set up Ruby
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 3.3.4
+          ruby-version: 2.5
+
+      - name: Install mechanize gem
+        run: |
+          gem install mechanize
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:24]: run bundle install ?

Comment (Dxuian) [reply .github/workflows/weekly_run.yml:24]: bundle is unncessary no ? 
gem is installing on its own  but ill try with bundle in the coming commits

Comment (Dxuian) [reply .github/workflows/weekly_run.yml:24]: beacuse run bundle install installs all the libs in the gemfile 
and we only need mechanize

File: .github/workflows/weekly_run.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+name: Weekly Ruby Script Runner
+
+on:
+  push:
+    branches:
+      - category
+  schedule:
+    - cron: '0 0 * * 0' # Runs every weekend
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:8]:
```suggestion
    - cron: '0 0 * * 0' # Runs every weekend
```

File: .github/workflows/weekly_run.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.3.4
+
+      - name: Install mechanize gem
+        run: |
+          gem install mechanize
+
+      - name: Run seek/t.rb
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:26]:
```suggestion
      - name: Run seek/t.rb
```

**Review by jbampton — COMMENTED** [review]
https://app.circleci.com/pipelines/github/slurpcode/slurp/8883/workflows/3beae9de-5501-42ec-b8c8-5a790b3774d9/jobs/70125

**Comment by jbampton** [comment]
Can you fix the rubocop errors ?  From `ci/circleci: build` step

https://app.circleci.com/pipelines/github/slurpcode/slurp/8909/workflows/8d532364-5356-4182-ac06-b4573f1f8766/jobs/70385

Autofix with `rubocop -A`

**Comment by Dxuian** [comment]
rubocop -A

**Comment by Dxuian** [comment]
> rubocop -A

done please review 🙏🙏🙏

**Comment by BaseMax** [comment]
This branch has conflicts that must be resolved
Use the [web editor](https://github.com/slurpcode/slurp/pull/2788/conflicts) or the  to resolve conflicts.
Conflicting files
seek/seek.rb

**Comment by Dxuian** [comment]
> This branch has conflicts that must be resolved Use the [web editor](https://github.com/slurpcode/slurp/pull/2788/conflicts) or the to resolve conflicts. Conflicting files seek/seek.rb

done 
pls review and merge

**Comment by Dxuian** [comment]
i wasnt familiar with manual text diffing , cuz i always used the inbuild differ from vs code i had no idea i was supposed to manually diff ?

File: seek/seek.rb [hunk 5]
```diff
@@ -35,7 +36,7 @@ def wtype(worktype)
     "242%2C243%2C244%2C245"
   end
 end
-
+git push --set-upstream origin fix-spelling
```
Comment (jbampton) [inline seek/seek.rb:39]:
```suggestion
```

**Comment by BaseMax** [comment]
Welcome back mate, John. happy to see your feedback again here.

**Comment by jbampton** [comment]
I fixed the conflicts now

File: .github/workflows/weekly_run.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
+name: Weekly Ruby Script Runner
+
+on:
+  # for testing purposes
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:4]:
```suggestion
  # for testing purposes
```

File: .github/workflows/weekly_run.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        run: |
+          git config --global user.name "github-actions[bot]";
+          git config --global user.email "github-actions[bot]@users.noreply.github.com";
+          git add seek/job_ind.json;
+          if git diff-index --quiet HEAD; then
+            echo "No changes to commit.";
+          else
+            git commit -m "Update seek/job_ind.json";
+            git push origin category;
+          fi;
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:52]:
```suggestion
          fi;

```

**Comment by Dxuian** [comment]
any changes ?

**Comment by jbampton** [comment]
@Dxuian can you please fix the `ci/circleci: build` step

https://app.circleci.com/pipelines/github/slurpcode/slurp/8916/workflows/f2160ff2-cd96-4775-a653-70aac7554cf7/jobs/70452

Some Rubocop errors

**Comment by jbampton** [comment]
https://github.com/slurpcode/slurp/blob/main/.rubocop.yml

https://docs.rubocop.org/rubocop/cops_metrics.html#metricsclasslength

@Dxuian you can add another entry to the `.rubocop.yml` for `Metrics/ClassLength` and set `Max` higher to stop this warning/errror.  Then there is no need to change the code

**Comment by jbampton** [comment]
Go to the next link and scroll to the bottom:

https://github.com/slurpcode/slurp/actions/runs/11242114988/job/31255094112?pr=2788

You need to "exclude": `seek/job_ind.json` from the `codespell` hook in the pre-commit config

**Comment by Dxuian** [comment]
> Go to the next link under heading 'Run pre-commit` and scroll to the bottom:
> 
> https://github.com/slurpcode/slurp/actions/runs/11242114988/job/31255094112?pr=2788
> 
> You need to "exclude": `seek/job_ind.json` from the `codespell` hook in the pre-commit config

done pls review 🙏🙏🙏

**Comment by jbampton** [comment]
`end-of-file-fixer` is failing on pre-commit

https://github.com/slurpcode/slurp/actions/runs/11242474382/job/31256225388?pr=2788

Can we create the JSON file with a blank line at the end ?

**Comment by Dxuian** [comment]
> > `end-of-file-fixer` is failing on pre-commit
> > https://github.com/slurpcode/slurp/actions/runs/11242474382/job/31256225388?pr=2788
> > Can we create the JSON file with a blank line at the end ?
> 
> done

@johnbampton  it appears the github actions bot is forcefully removing the newline that we added for avoiding failing the precommit check

**Comment by Dxuian** [comment]
@johnbampton  ive managed to fix the problem

**Comment by Dxuian** [comment]
please review and merge 🙏🙏🙏

File: .github/workflows/weekly_run.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
+name: Weekly Ruby Script Runner
+
+on:
+  # for testing purposes
+  push:
+    branches:
+      - category
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:7]:
```suggestion
      - main
```

??

File: .github/workflows/weekly_run.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        if: env.exit_code == '0'
+        run: |
+          git config --global user.name "github-actions[bot]";
+          git config --global user.email "github-actions[bot]@users.noreply.github.com";
+          git add seek/job_ind.json;
+          if git diff-index --quiet HEAD; then
+            echo "No changes to commit.";
+          else
+            git commit -m "Update seek/job_ind.json";
+            git push origin category;
```
Comment (jbampton) [inline .github/workflows/weekly_run.yml:51]:
```suggestion
            git push origin main;
```

??

**Comment by jbampton** [comment]
> i did not think it would take this long 😢😢😢😭😭 when i picked up the issue

Was marked as a hard issue:

https://github.com/slurpcode/slurp/issues/2145

**Comment by Dxuian** [comment]
any other changes ?

File: .rubocop.yml [hunk 10]
```diff
@@ -68,3 +68,6 @@ Style/StringLiterals:
 
 Style/StringLiteralsInInterpolation:
   Enabled: true
+
+Metrics/ClassLength:
+  Max: 120
```
Comment (jbampton) [inline .rubocop.yml:73]: All cops should be ordered.  There is a `metrics` section.

**Comment by Dxuian** [comment]
it passed the relevant tests finally , any changes ?

**Review by jbampton — APPROVED** [review]

**Comment by Dxuian** [comment]
well its finally over

**Pull request merged**